tc/core/polyhedral/scop.cc

```diff
@@ -440,6 +440,89 @@ detail::ScheduleTree* obtainOuterBand(detail::ScheduleTree* root) {
   }
   return tree;
 }
+
+// Check if there is at least one band with a coincident dimension
+// in the tree.
+bool hasAtLeastOneCoincidentLoop(detail::ScheduleTree* tree) {
+  if (auto band = tree->elemAs<ScheduleTreeElemBand>()) {
+    if (find(band->coincident_.begin(), band->coincident_.end(), true) !=
+            band->coincident_.end() &&
+        band->permutable_) {
+      return true;
+    }
+  }
+  auto n = tree->numChildren();
+  if (n == 0) {
+    return false;
+  } else if (n == 1) {
+    return hasAtLeastOneCoincidentLoop(tree->child({0}));
```

Comment:
If this function really wants to implement `hasAtLeastOneCoincidentLoop` along at least one path then by all means let's drop recursion and go a collect followed by a test (in which case, hoist the code above in a separate helper function).

However, see my comment in  `obtainOuterCoincidentBands` I am wondering if you want an "along all paths" property?